packages/teleterm/src/ui/services/quickInput/types.ts

```diff
@@ -34,4 +36,17 @@ export type Item =
   | ItemDb
   | ItemCluster
   | ItemCmd
+  | ItemSshLogin
   | ItemEmpty;
+
+type AutocompleteBase<T> = {
+  kind: T;
+  picker: QuickInputPicker;
+};
+
+export type AutocompletePartialMatch =
+  AutocompleteBase<'autocomplete.partial-match'> & { listItems: Item[] };
+
+export type AutocompleteNoMatch = AutocompleteBase<'autocomplete.no-match'>;
+
+export type AutocompleteResult = AutocompletePartialMatch | AutocompleteNoMatch;
```

Comment:
So initially this had like four or five different possible values. I changed my approach twice and what we're left with is clearly just a binary choice which we could represent trough just `null | AutocompletePartialMatch`. However, I'm not 100% sure how actually picking the suggestions will change this, but I'll be fine with changing it in the future.

Now that I think about it though, we'll always have to return the picker anyway. But then `listItems` can be just `null | Item[]`.